pkg/dao/rpm.go

```diff
@@ -116,3 +116,50 @@ func (r rpmDaoImpl) modelToApiFields(in *models.Rpm, out *api.RepositoryRpm) {
 	out.Summary = in.Summary
 	out.Checksum = in.Checksum
 }
+
+func (r rpmDaoImpl) Search(orgID string, request api.SearchRpmRequest, limit int) ([]api.SearchRpmResponse, error) {
+	// Retrieve the repository id list
+	if orgID == "" {
+		return nil, fmt.Errorf("orgID can not be an empty string")
+	}
+	if len(request.URLs) == 0 {
+		return nil, fmt.Errorf("request.URLs must contain at least 1 URL")
+	}
```

Comment:
I could see there being some confusion down the road around trialing slashes.  I.e. if we have a repository:
http://example.com/foo/

but someone searches for  http://example.com/foo

and gets no results.  Or the other way around.  Maybe we should check for both?  


We should probably also streamline the repository creation to only create with trailing slashes?  So that a user can't create http://example.com/foo/ & http://example.com/foo by mistake.     I'll file an issue for the creation/update story, but we could go ahead and handle searches for both here, thoughts?

Reply:
:+1: for normalizing the url into the database! I'd rather a cleaner query, so I would add a mark `// FIXME 103` and a description so we can locate it and update it in the future

Reply:
I have added a quick pre-process to the handler, so `http://example.com/foo/` will be converted to `http://example.com/foo`; I have added the above mark for future change.